plugins/inputs/ravendb/ravendb.go

```diff
@@ -108,40 +97,39 @@ func (r *RavenDB) Gather(acc telegraf.Accumulator) error {
 
 	var wg sync.WaitGroup
 
-	if r.GatherServerStats {
-		wg.Add(1)
-
-		go func() {
-			defer wg.Done()
-			r.gatherServer(acc)
-		}()
-	}
-
-	if r.GatherDbStats {
-		wg.Add(1)
-
-		go func() {
-			defer wg.Done()
-			r.gatherDatabases(acc)
-		}()
-	}
-
-	if r.GatherIndexStats {
-		wg.Add(1)
-
-		go func() {
-			defer wg.Done()
-			r.gatherIndexes(acc)
-		}()
-	}
-
-	if r.GatherCollectionStats {
-		wg.Add(1)
-
-		go func() {
-			defer wg.Done()
-			r.gatherCollections(acc)
-		}()
+	for _, statToCollect := range r.StatsInclude {
+		switch statToCollect {
+		case "server":
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				r.gatherServer(acc)
+			}()
+		case "databases":
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				r.gatherDatabases(acc)
+			}()
+		case "indexes":
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				r.gatherDatabases(acc)
+			}()
+		case "collections":
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				r.gatherCollections(acc)
+			}()
+		default:
+			panic(fmt.Sprintf("unhandled statistics type %s", statToCollect))
```

Comment:
Please don't panic here! You can still `acc.AddError()`.

Reply:
done